crates/compiler/build/src/program.rs

```diff
@@ -756,121 +756,40 @@ fn build_loaded_file<'a>(
     code_gen_options: CodeGenOptions,
     emit_timings: bool,
     link_type: LinkType,
-    mut linking_strategy: LinkingStrategy,
-    prebuilt_requested: bool,
+    linking_strategy: LinkingStrategy,
+    build_host_requested: bool,
     wasm_dev_stack_bytes: Option<u32>,
     loaded: roc_load::MonomorphizedModule<'a>,
     compilation_start: Instant,
     out_path: Option<&Path>,
 ) -> Result<BuiltFile<'a>, BuildFileError<'a>> {
+    // get the platform path from the app header
     let platform_main_roc = match &loaded.entry_point {
         EntryPoint::Executable { platform_path, .. } => platform_path.to_path_buf(),
         _ => unreachable!(),
     };
 
-    // For example, if we're loading the platform from a URL, it's automatically prebuilt
-    // even if the --prebuilt-platform CLI flag wasn't set.
-    let is_platform_prebuilt = prebuilt_requested || loaded.uses_prebuilt_platform;
```

Comment:
I should add a check here; if `loaded.uses_prebuilt_platform` is true and also `build_host_requested` then the platform is a URL release and we have also been instructed to rebuild the host. These two are incompatible because roc cannot build the host.